### PR TITLE
Event fields are not optional.

### PR DESCRIPTION
**Pull request by andy-faunadb**

### CODE REVIEW

File: faunadb/objects.py [hunk 1]
```diff
@@ -95,19 +95,18 @@ def from_raw(raw):
...
-  def __init__(self, ts, action=None, resource=None):
+  def __init__(self, ts, action, resource):
+    if action not in ("create", "delete"):
+      raise ValueError("Action must be \"create\" or \"delete\".")
     self.ts = ts
     "Microsecond UNIX timestamp at which the event occurred."
-    if action not in (None, "create", "delete"):
-      raise ValueError("Action must be create or delete or None.")
     self.action = action
     """"create" or "delete"""""
```
Comment (ashfire908) [inline faunadb/objects.py]: Looks like there's one too many double quotes here.

File: faunadb/objects.py [hunk 2]
```diff
@@ -95,19 +95,18 @@ def from_raw(raw):
...
     self.resource = resource
-    "The Ref of the affected instance."
+    """The Ref of the affected instance."""
```
Comment (ashfire908) [inline faunadb/objects.py]: What's the reason for going triple double quotes here?

File: faunadb/objects.py [hunk 3]
```diff
@@ -95,19 +95,18 @@ def from_raw(raw):
     return Event(raw["ts"], raw["action"], raw["resource"])
 
   # pylint: disable=invalid-name
-  def __init__(self, ts, action=None, resource=None):
+  def __init__(self, ts, action, resource):
+    if action not in ("create", "delete"):
+      raise ValueError("Action must be \"create\" or \"delete\".")
```
Comment (ashfire908) [inline faunadb/objects.py]: You can avoid the escaping by just using single quotes here.

File: faunadb/objects.py (hunk 2 above)
Comment (andy-faunadb) [inline faunadb/objects.py]: This is a property docstring. Not accessible at runtime, but sphinx uses it.

**Comment by ashfire908** [comment]
LGTM

**Pull request merged**